### ISSUE TITLE
Event streaming is broken when minio is running behind a Traefik proxy

### ISSUE THREAD

**Issue by Rio** [minio/minio-go#731]
It seems that events get buffered in [Traefik](https://traefik.io). So only when the buffer is full it gets flushed to the event listening client, that could either be `mc watch` or `minio.Client.ListenBucketNotification`.

Here is a `docker-compose.yml` file to demonstrate the issue by doing a simple `docker-compose up` to start the services.
```yaml
version: "2"

services:
  traefik:
    image: traefik
    command:
      - "--web" # Enable the traefik web interface available on localhost:8080
      - "--docker" # Enable pulling configuration out of the docker daemon
      - "--docker.domain=localhost" # Set the default domain to localhost
    ports:
      - "80:80" # Expose the http port where traefik listens on
      - "8080:8080" # Expose the port traefik's web interface is served on
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock" # Mount the docker socket inside the container so traefik can communicate with it. Note, could have SELinux implications.

  minio:
    image: minio/minio
    command:
      - "server"
      - "/data"
    labels:
      - "traefik.port=9000" # Tell traefik to route to port 9000
      - "traefik.frontend.rule=Host:localhost" # Tell traefik to route to minio when a request is made on localhost:80
    environment:
      MINIO_ACCESS_KEY: testaccesskey
      MINIO_SECRET_KEY: testsecretkey
    ports:
      - "9000:9000" # Expose minio's port directly so it is also available on localhost:9000
```

After running `docker-compose up` you can use the `examples/minio/listenbucketnotification.go` file to connect to `localhost:9000` for a direct connection to minio or to `localhost:80` for a connection to minio through the traefik proxy both with TLS disabled. You could also use `mc watch` on both endpoints to see the results simultaneously. The minio web interface is available by going to `localhost`.

After making a bucket and setting up the listeners or watches on it you can start copying files to minio through either endpoint or the web interface. You'll see the listener that listens on `localhost:9000` respond immediately but the listener that goes to the proxy using `localhost:80` will only get all the events at the same time after about 10 copies.

As @vadmeste pointed out it could have something to do with the `Content-Type` header on the listen not being set to `text/event-stream`, see this issue on the Traefik github https://github.com/containous/traefik/issues/560.

y4m4 also pointed us to the MDN docs for [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events).

**Comment by deekoder** [comment]
Will get back to you @Rio

**Comment by harshavardhana** [comment]
This is fixed in master branch of minio by PR #4621
